governance/program/src/processor/process_sign_off_proposal.rs

```diff
@@ -41,7 +38,9 @@ pub fn process_sign_off_proposal(program_id: &Pubkey, accounts: &[AccountInfo])
     proposal_data.assert_can_sign_off()?;
 
     // If the owner of the proposal hasn't appointed any signatories then can sign off the proposal themself
-    if proposal_data.signatories_count == 0 {
+    if proposal_data.signatories_count == 0
+        && proposal_data.signatories_count >= governance_data.required_signatories_count
```

Comment:
Since the first condition checks for `proposal_data.signatories_count == 0` the only way for the second condition to be true is for `governance_data.required_signatories_count == 0` which would be more explicit.

However I think it's still wrong because we want a strong and explicit guarantee here to ensure the signing off process can't start before all required signatories are added and once it starts then the Proposals enters `SigningOff` state and signatories can't be modified any longer. 

You can add this condition:

```
    if (governance_data.required_signatories_count > 0
        && proposal_data.signatories_count < governance_data.required_signatories_count)
    {
        return Err(GovernanceError::MissingRequiredSignatories.into());
    }
```

And then you can remove the extra checks in line `42` & `89`

Reply:
You're right, that's a much better way to think about it. Resolved.